services/catalog/rs_server_catalog/user_catalog.py

```diff
@@ -318,7 +318,10 @@ def find_owner_id(self, ecql_ast: Node) -> str:
                 res = right
         return res
 
-    async def manage_search_request(self, request: Request) -> Request | JSONResponse:
+    async def manage_search_request(  # pylint: disable = too-many-branches
```

Comment:
Try to simplify this, there are too many decisional branches. Keep it simple for MC/DC